src/main/java/ulcrs/data/DataFetch.java

```diff
@@ -0,0 +1,80 @@
+package ulcrs.data;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+class DataFetch {
+
+    // Constants for request
+    private final static String DROP_IN_REQUEST_URL = "https://dropin-dev.engr.wisc.edu/services/DIMainController.php";
+    private final static String DROP_IN_REQUEST_DATA = "service=ScheduleService&function=exportSchedulerInformation";
+    private final static String DROP_IN_REQUEST_REQUEST_TYPE = "POST";
+    private final static String DROP_IN_REQUEST_CONTENT_TYPE = "application/x-www-form-urlencoded";
+    private final static String DROP_IN_REQUEST_CONTENT_LANGUAGE = "en-US";
+    private final static String CONTENT_TYPE = "Content-Type";
+    private final static String COOKIE = "Cookie";
+    private final static String CONTENT_LENGTH = "Content-Length";
+    private final static String CONTENT_LANGUAGE = "Content-Language";
+
+    /*
+    This hits the Drop-In endpoint to retrieve course, shift, tutor, and tutor preference data. However, Drop-In has a peculiar way of doing requests, so we issue a HTTP POST to the following:
+        • URL: https://dropin-dev.engr.wisc.edu/services/DIMainController.php
+        • Set Content-Type to application/x-www-form-urlencoded
+        • Set request data to: service=ScheduleService&function=exportSchedulerInformation
+        • You must include the shibboleth cookie/credentials (passed as String)
+        • You must have admin credentials on the dev website (I believe Matt and Jason have this).
+    */
+    static List<String> fetchFromULCServer(String cookie) {
+        HttpURLConnection connection = null;
+
+        try
+        {
+            // Initialize HTTP request
+            URL url = new URL(DROP_IN_REQUEST_URL);

```

Comment:
We could change URL to the more specific HttpRequest in the future

Reply:
I agree. This is largely copypasta from Will's test, so I intend to investigate this more for a better solution. Sounds like you already know it!